runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,226 @@
-const { json, send } = require("micro");
+const { text, json, send } = require("micro");
 
 const USER_CODE_PATH = '/usr/code-start';
 
 const server = micro(async (req, res) => {
-    const body = await json(req);
-
-    if (req.headers[`x-internal-challenge`] !== process.env['INTERNAL_RUNTIME_KEY']) {
-        return send(res, 500, {stderr: 'Unauthorized'});
+    if (req.headers[`x-open-runtimes-secret`] !== process.env['INTERNAL_RUNTIME_KEY']) {
+        return send(res, 500, 'Unauthorized');
     }
 
-    const request = {
-        variables: body.variables ?? {},
-        headers: body.headers ?? {},
-        payload: body.payload ?? '',
+    const logs = [];
+    const errors = [];
+    const response = {
+        headers: {},
+        body: '',
+        statusCode: 200,
+
+        cookies: {},
+        contentType: null,
     };
 
-    console.stdlog = console.log.bind(console);
-    console.stderror = console.error.bind(console);
-    console.stdinfo = console.info.bind(console);
-    console.stddebug = console.debug.bind(console);
-    console.stdwarn = console.warn.bind(console);
+    const contentType = req.headers['content-type'] ?? 'text/plain';
+    const body = contentType === 'application/json' ? await json(req) : await text(req);
 
-    const logs = [];
-    const errors = [];
+    const headers = {};
+    Object.keys(req.headers).filter((header) => !header.startsWith('x-open-runtimes-')).forEach((header) => {
+        headers[header] = req.headers[header];
+    });
 
-    console.log = console.info = console.debug = console.warn = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        logs.push(args.join(" "));
-    }
+    const context = {
+        req: {
+            body: body,
+            headers: headers,
+            method: req.method,
+            url: req.url
+        },
+        res: {
+            getBody: function() {
+                return response.body;
+            },
 
-    console.error = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        errors.push(args.join(" "));
-    }
+            getStatusCode: function() {
+                return response.statusCode;
+            },
 
-    const response = {
-        send: (text, status = 200) => send(res, status, {response: text, stdout: logs.join('\n'), stderr: errors.join('\n')}),
-        json: (json, status = 200) => send(res, status, {response: json, stdout: logs.join('\n'), stderr: errors.join('\n')}),
+            getContentType: function() {
+                return response.contentType
```

Comment:
IIRC there can be a `content-type` header in both a request and response.

In a request, the `content-type` header tells the the server the type of the request.
In a response, the `content-type` headers tells the browser or resulting client the type of the response. 

Don't we need to have getters for both the content types ?